frame/nomination-pools/src/mock.rs

```diff
@@ -20,17 +19,19 @@ pub fn default_reward_account() -> AccountId {
 parameter_types! {
 	pub static CurrentEra: EraIndex = 0;
 	pub static BondingDuration: EraIndex = 3;
-	static BondedBalanceMap: HashMap<AccountId, Balance> = Default::default();
-	static UnbondingBalanceMap: HashMap<AccountId, Balance> = Default::default();
+	pub storage BondedBalanceMap: BTreeMap<AccountId, Balance> = Default::default();
```

Comment:
@shawntabrizi @KiChjang @ggwpez this took me a bit of time to realize and fix, so want to inform you all about it: with making things transactional by default (which we also use in pools), using statics is a bit tricky because they do not revert.

Reply:
Statics produce other problems as well (eg. https://github.com/paritytech/substrate/issues/10479), I think getting rid of them is good.

Reply:
There are some scenarios where `storage` cannot be used, like when there is no externalities environment. Could be all of these are fixed by creating an exernalities wrapper, but I dont currently see that all statics can be removed.